### PR TITLE
Delta onboarding logic (2665 - 2761)

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
This PR introduces a solution to have version depending delta onboardings after the initial onboarding.

Story: [2761](https://jira.itc.sap.com/browse/EXPOSUREAPP-2761)

## Checklist

* [x] This PR does not change text that hasn't been signed off with the RKI. Have a look at the pinned issue [440](https://github.com/corona-warn-app/cwa-app-ios/issues)
* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed.
* [ ] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	var appVersion: String {
+		guard let version = infoDictionary?["CFBundleShortVersionString"] as? String else {
+			fatalError("Could not read CFBundleShortVersionString from Bundle.")
+		}
+		return version
+	}
+
+	var appBuildNumber: String {
+		guard let buildNumber = infoDictionary?["CFBundleVersion"] as? String else {
```
Comment (AppDevelopersDE) [inline src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift:30]:
```suggestion
		guard let buildNumber = infoDictionary?[kCFBundleVersionKey as String] as? String else {
```
better use system provided key ?

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Extensions/Bundle+Version.swift:30]: Nice!

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift [hunk 2]
```diff
@@ -60,13 +60,9 @@ extension AppInformationViewController {
...
-		if let bundleVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"],
-			let bundleBuild = Bundle.main.infoDictionary?["CFBundleVersion"] {
-			versionLabel.text = "\(AppStrings.Home.appInformationVersion) \(bundleVersion) (\(bundleBuild))"
-		} else {
-			versionLabel.text = "\(AppStrings.Home.appInformationVersion) <unknown>"
-			logError(message: "Unknown version. Should not happen!")
-		}
+		let bundleVersion = Bundle.main.appVersion
+		let bundleBuild = Bundle.main.appBuildNumber
+		versionLabel.text = "\(AppStrings.Home.appInformationVersion) \(bundleVersion) (\(bundleBuild))"
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:65]: Shouldn't that be part of a ViewModel?

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingCoordinatorTests.swift [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	) {
+		self.version = version
+		self.store = store
+		self.deltaViewController = deltaViewController
+	}
+
+	func makeViewController() -> DeltaOnboardingViewController {
+		return deltaViewController
+	}
+}
```
Comment (SwissMountainFrog) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingCoordinatorTests.swift:168]: I really like these tests 🚀

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:65]: Yes you are right, but there is no view model. Lets add AppInformationViewController to the refactoring list.

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingCoordinatorTests.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/__tests__/DeltaOnboardingCoordinatorTests.swift:168]: ❤️

**Review by SwissMountainFrog — APPROVED** [review]

**Review by KaiTeuber — APPROVED** [review]

**Pull request merged**